### PR TITLE
Remove repeated work, add test for v2 subtypes in cards.

### PR DESCRIPTION
**Pull request by plural**
midnight sun cards had 2 new subtypes (harmonic and industrial).  The new API server card importer broke.

### CODE REVIEW

File: test/validate_v2.ts [hunk 1]
```diff
@@ -120,6 +120,18 @@ describe('Cards', () => {
       }
     });
   });
+
+  it('have valid subtypes', () => {
+    const subtypes = new Set<string>(getCardSubtypesV2Json().map(c => c.id));
+    cardFiles.forEach(file => {
+      const card = JSON.parse(fs.readFileSync(resolve(cardDir, file), 'utf-8'));
```
Comment (NoahTheDuke) [inline test/validate_v2.ts:127]: I just realized, you're reading and parsing the files in every test? Maybe move this line onto a before (not beforeEach) block or even just directly into a top-level const.

File: test/validate_v2.ts [hunk 2]
```diff
@@ -120,6 +120,18 @@ describe('Cards', () => {
...
+      if ('subtypes' in card) {
+        card.subtypes.forEach(s => {
```
Comment (NoahTheDuke) [inline test/validate_v2.ts:129]: I think these can be condensed with [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining): `card.subtypes?.forEach`

File: test/validate_v2.ts [hunk 3]
```diff
@@ -120,6 +120,18 @@ describe('Cards', () => {
       }
     });
   });
+
+  it('have valid subtypes', () => {
+    const subtypes = new Set<string>(getCardSubtypesV2Json().map(c => c.id));
```
Comment (NoahTheDuke) [inline test/validate_v2.ts:125]: Could/should be a top level const too.

File: test/validate_v2.ts (hunk 2 above)
Comment (plural) [reply test/validate_v2.ts:129]: wat?!?!  that's very cool.  thanks, Noah.

**Comment by plural** [comment]
Ok, i moved ~all the file reads that were repeated outside the test cases.  This saved a bit of time and removed some code.

Before:
```
  78 passing (431ms)
  78 passing (433ms)
  78 passing (422ms)
  78 passing (437ms)
  78 passing (419ms)
  78 passing (418ms)
  78 passing (416ms)
  78 passing (426ms)
  78 passing (441ms)
  78 passing (418ms)
  78 passing (412ms)
  78 passing (418ms)
  78 passing (421ms)
  78 passing (445ms)
  78 passing (447ms)
  78 passing (452ms)
  78 passing (426ms)
  78 passing (421ms)
  78 passing (428ms)
  78 passing (436ms)
```
After:
```
jasongessner@Jasons-MacBook-Pro netrunner-cards-json (subtype-tests)*$ for NUM in `seq 1 100`; do npm run test | grep ' passing'; done
  79 passing (376ms)
  79 passing (392ms)
  79 passing (374ms)
  79 passing (376ms)
  79 passing (373ms)
  79 passing (372ms)
  79 passing (387ms)
  79 passing (377ms)
  79 passing (385ms)
  79 passing (380ms)
  79 passing (381ms)
  79 passing (371ms)
  79 passing (385ms)
  79 passing (377ms)
  79 passing (377ms)
  79 passing (424ms)
```

**Review by NoahTheDuke — APPROVED** [review]
Hell yeah, this looks great!

**Pull request merged**